lib/credit_card_sanitizer.rb

```diff
@@ -170,7 +172,22 @@ def is_tracking?(candidate, options)
   end
 
   def valid_numbers?(candidate, options)
-    LuhnChecksum.valid?(candidate.numbers) && valid_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+    LuhnChecksum.valid?(candidate.numbers) && valid_company_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+  end
+
+  def valid_context?(options)
+    return true if valid_prefix?(options[:prefix]) && valid_postfix?(options[:postfix])
+    !options[:parse_flanking]
+  end
+
+  def valid_prefix?(prefix)
+    return true if prefix.nil? || prefix.downcase.end_with?(*ACCEPTED_PREFIX)
```

Comment:
@ggrossman brought up a good point that using `downcase` as I suggested isn't a great idea here.